### PR TITLE
Feature: Disable show more options menu item until shell items are loaded

### LINKED ISSUE #11236: Feature: Make context menu entry "Show more options" always visible
### What feature or improvement do you think would benefit Files?

The asynchronous loading of the shell extensions when opening the context menu still leads to a rather ugly "jumping" or "expansion" of the context menu UI as the entry "Show more options" gets added with some delay.

Now, the latest improvements made in #11110 make the situation already better than previously. 
May it be possible to just always (instantly) display the "show more options" entry, so that the context menu can completely get rid of the jumping / expansion? My idea is that the entry "Show more options" could be grayed out (in UI design terms "disabled") initially when opening the context menu and just gets enabled once the shell extensions have been loaded.

Related: #11112 

### Requirements

- This proposal will accomplish making the entry "Show more options" in the context menu always visible
- This proposal will accomplish disabling the entry initially before the shell extensions have finished loading

### Files Version

2.4.33

### Windows Version

Windows 11 22H2 Build 22621.1194

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #11236

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Comment by yaira2** [comment]
@hecksmosis can you fix the merge conflicts?

**Comment by hecksmosis** [comment]
Done

File: src/Files.App/BaseLayout.cs [hunk 1]
```diff
@@ -585,14 +585,14 @@ private async Task LoadMenuItemsAsync()
 			shellContextMenuItemCancellationToken = new CancellationTokenSource();
 			SelectedItemsPropertiesViewModel.CheckAllFileExtensions(SelectedItems!.Select(selectedItem => selectedItem?.FileExtension).ToList()!);
 			var shiftPressed = Microsoft.UI.Input.InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Shift).HasFlag(Windows.UI.Core.CoreVirtualKeyStates.Down);
-			var items = ContextFlyoutItemHelper.GetItemContextCommandsWithoutShellItems(currentInstanceViewModel: InstanceViewModel!, selectedItems: SelectedItems!, selectedItemsPropertiesViewModel: SelectedItemsPropertiesViewModel, commandsViewModel: CommandsViewModel!, shiftPressed: shiftPressed);
+			var items = ContextFlyoutItemHelper.GetItemContextCommandsWithoutShellItems(currentInstanceViewModel: InstanceViewModel!, selectedItems: SelectedItems!, selectedItemsPropertiesViewModel: SelectedItemsPropertiesViewModel, commandsViewModel: CommandsViewModel!, shiftPressed: shiftPressed);
```
Comment (ferrariofilippo) [inline src/Files.App/BaseLayout.cs:588]: Can you remove trailing space?

File: src/Files.App/BaseLayout.cs [hunk 2]
```diff
@@ -585,14 +585,14 @@ private async Task LoadMenuItemsAsync()
...
-			var items = ContextFlyoutItemHelper.GetItemContextCommandsWithoutShellItems(currentInstanceViewModel: InstanceViewModel!, selectedItems: SelectedItems!, selectedItemsPropertiesViewModel: SelectedItemsPropertiesViewModel, commandsViewModel: CommandsViewModel!, shiftPressed: shiftPressed);
+			var items = ContextFlyoutItemHelper.GetItemContextCommandsWithoutShellItems(currentInstanceViewModel: InstanceViewModel!, selectedItems: SelectedItems!, selectedItemsPropertiesViewModel: SelectedItemsPropertiesViewModel, commandsViewModel: CommandsViewModel!, shiftPressed: shiftPressed);			
 			ItemContextMenuFlyout.PrimaryCommands.Clear();
 			ItemContextMenuFlyout.SecondaryCommands.Clear();
 			var (primaryElements, secondaryElements) = ItemModelListToContextFlyoutHelper.GetAppBarItemsFromModel(items);
 			AddCloseHandler(ItemContextMenuFlyout, primaryElements, secondaryElements);
-			primaryElements.ForEach(i => ItemContextMenuFlyout.PrimaryCommands.Add(i));
+      primaryElements.ForEach(ItemContextMenuFlyout.PrimaryCommands.Add);
```
Comment (ferrariofilippo) [inline src/Files.App/BaseLayout.cs:593]: Spacing seems off

File: src/Files.App/BaseLayout.cs [hunk 3]
```diff
@@ -692,10 +692,12 @@ private void AddShellItemsToMenu(List<ContextMenuFlyoutItemViewModel> shellMenuI
...
 
-					if (overflowItemFlyout.Items.Count > 0)
+					if (overflowItemFlyout.Items.Count > 0 && UserSettingsService.AppearanceSettingsService.MoveShellExtensionsToSubMenu)
+					{ 
+						overflowItem.IsEnabled = true; 
+					} else if (!UserSettingsService.AppearanceSettingsService.MoveShellExtensionsToSubMenu)
 					{
-						(contextMenuFlyout.SecondaryCommands.First(x => x is FrameworkElement fe && fe.Tag as string == "OverflowSeparator") as AppBarSeparator)!.Visibility = Visibility.Visible;
-						overflowItem.Visibility = Visibility.Visible;
+						overflowItem.Visibility = Visibility.Collapsed;
```
Comment (ferrariofilippo) [inline src/Files.App/BaseLayout.cs:700]: `else if` should be aligned on a new line. You can remove brackets

File: src/Files.App/UserControls/SidebarControl.xaml.cs [hunk 4]
```diff
@@ -936,6 +936,66 @@ private void ResizeElementBorder_ManipulationStarted(object sender, Manipulation
...
+		private async void LoadShellMenuItems(CommandBarFlyout itemContextMenuFlyout, ContextMenuOptions options)
+		{
+			try
+			{
+				if (options.ShowEmptyRecycleBin)
+				{
+					var emptyRecycleBinItem = itemContextMenuFlyout.SecondaryCommands.FirstOrDefault(x => x is AppBarButton appBarButton && (appBarButton.Tag as string) == "EmptyRecycleBin") as AppBarButton;
+					if (emptyRecycleBinItem is not null)
+					{
+						var binHasItems = RecycleBinHelpers.RecycleBinHasItems();
```
Comment (ferrariofilippo) [inline src/Files.App/UserControls/SidebarControl.xaml.cs:948]: Can you use the helper you introduced with #11204 ?

Comment (hecksmosis) [reply src/Files.App/UserControls/SidebarControl.xaml.cs:948]: I just noticed this is unused, I'll remove it altogether

**Comment by ferrariofilippo** [comment]
Works totally fine for folders and files, but not for sidebar items. Also, you should implement this for widgets as well.

**Comment by hishitetsu** [comment]
One suggestion. Instead of simply disabling the menu while loading, how about changing the text to, for example, "Loading more options"? Simply disabling the menu may lead people to think that no additional options exist.

**Comment by yaira2** [comment]
> One suggestion. Instead of simply disabling the menu while loading, how about changing the text to, for example, "Loading more options"? Simply disabling the menu may lead people to think that no additional options exist.

I agree

**Comment by yaira2** [comment]
Should this be applied to the widgets as well?

**Comment by hecksmosis** [comment]
Probably

**Comment by yaira2** [comment]
I added the loading label for base layout but it still needs to be added to the sidebar and widgets.

**Comment by yaira2** [comment]
@hecksmosis with the loading label, should we still disable the menu or is the loading label clear enough?

**Comment by hishitetsu** [comment]
I think it is still better to disable the menu.

**Comment by yaira2** [comment]
Widgets are done 👍

**Comment by yaira2** [comment]
One thing I noticed is that right clicking on Recycle Bin (from the Quick Access widget) or any file in the Recent Files widget causes the app to hang.

**Review by yaira2 — APPROVED** [review]

**Comment by hecksmosis** [comment]
Thanks for finishing it, had a pretty busy couple of days and couldn't work on this

**Comment by hishitetsu** [comment]
> One thing I noticed is that right clicking on Recycle Bin (from the Quick Access widget) or any file in the Recent Files widget causes the app to hang.

I will submit a PR to fix these issues.

**Pull request merged**